storagenode/contact/chore.go

```diff
@@ -48,53 +58,108 @@ func (chore *Chore) Run(ctx context.Context) (err error) {
 	defer mon.Task()(&ctx)(&err)
 	chore.log.Info("Storagenode contact chore starting up")
 
+	var group errgroup.Group
+
 	if !chore.service.initialized.Wait(ctx) {
 		return ctx.Err()
 	}
+	chore.mu.Lock()
+	for _, satellite := range chore.trust.GetSatellites(ctx) {
```

Comment:
I'm not sure if this will be compatible with real-time editing of the trusted satellites. If I understand this correctly, when we call `contact.chore.run` from the peer file, it will iterate over the satellites once. Then, each `cycle.Start` will execute the passed function on the interval as long as a nil error is returned

Reply:
Yeah you're right here. At the moment it won't work with dynamic modifications of the trusted list. I think that's ok for now, and we have another ticket here to add this capability once the overall design is complete. https://storjlabs.atlassian.net/browse/V3-3002